### DISCUSSION TITLE
Animation death for npcs exclusively

### DISCUSSION THREAD

**Discussion by WeylonSantana** [AscensionGameDev/Intersect-Engine#1497] [Ideas]
**Is your feature request related to a problem? Please describe.**
Well, it's something that has already been asked a lot, I just came to formalize, because of something I saw on Intersect.

**Describe the solution you'd like**
I know it has nothing to do with it, but the resources have death animation, so putting an animation tab in the npc window would be nice. And it doesn't have to be a death animation precisely, but the ability to reproduce an animation at the npc's death place. As an event. The only reason events cannot reproduce ("simulate") an animation of death is because they don't know where the npc died. The animation would be recorded in the animation editor and played at the npc's death site.

**Additional context**

**Comment by Cheshire92** [comment]
I don't think an animation playing on death for NPCs is too unreasonable.
Once the other editor changes get merged I'll take a look.

**Comment by WeylonSantana** [comment]
bringing extra information from #1460
![image](https://user-images.githubusercontent.com/63019821/192093407-8745c95e-0eaa-4b77-bf10-59fd584e279f.png)

**Comment by TimtyG** [comment]
Bumping this discussion.

I think it would be a nice touch to display the player or NPC sprite upon death. For example, when a player or NPC dies, their sprite will change to indicate their death, and we could hold the entity in that state for a short period before respawning.

It should also be configurable, allowing the user to decide whether they want this feature enabled or not.

I'd be happy to assist with the implementation if needed.

**Reply by Arufonsu** [reply]
i'd like the idea that we discussed along at some point in discord too, for example: we could optionally allow it to use the item drop handler to display an NPC's "corpse" (instead of items directly) after their death, so when we get close to the corpse, we would see the loot window.

![{FAB0AEBB-7B6E-488E-8449-EA5AE4417B6A}](https://github.com/user-attachments/assets/11f603f9-e926-466d-b99e-06c6a124f033)

![{92436169-5A59-4557-876C-318D8CAC8515}](https://github.com/user-attachments/assets/44dff08a-350f-4b5d-b706-de576a63e948)


if anything, this and other ideas should be optional/switchable with some enum in mind for configs
